src/_igraph/graphobject.c

```diff
@@ -5017,80 +5016,65 @@ PyObject *igraphmodule_Graph_get_shortest_paths(igraphmodule_GraphObject *
     return NULL;
   }
 
+  /* The starting point is a single vertex, but the end is a list
+   * of vertices, so we need one shortest path per target vertex */
   if (igraph_vs_size(&self->g, &to, &no_of_target_nodes)) {
     if (weights) { igraph_vector_destroy(weights); free(weights); }
     igraph_vs_destroy(&to);
     igraphmodule_handle_igraph_error();
     return NULL;
   }
 
-  ptrvec = (igraph_vector_ptr_t *) calloc(1, sizeof(igraph_vector_ptr_t));
-  if (!ptrvec) {
+  /* Memory for the pointer */
+  veclist = (igraph_vector_int_list_t *) calloc(1, sizeof(igraph_vector_int_list_t));
```

Comment:
I think we don't need to allocate `veclist` on the heap, we can simply use an `igraph_vector_int_list_t`. I have no idea why the original implementation allocated `ptrvec` on the heap; it must have been one of the first functions involving `igraph_vector_ptr_t` that I converted from C to Python and I think I have overcomplicated it.

Reply:
yep ok